common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -748,11 +747,37 @@ public static Inet6Address getByName(CharSequence ip) {
      * @param ipv4Mapped
      * <ul>
      * <li>{@code true} To allow IPv4 mapped inputs to be translated into {@link Inet6Address}</li>
-     * <li>{@code false} Don't turn IPv4 addressed to mapped addresses</li>
+     * <li>{@code false} Consider IPv4 mapped addresses as invalid.</li>
      * </ul>
      * @return {@link Inet6Address} representation of the {@code ip} or {@code null} if not a valid IP address.
      */
     public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
+        byte[] bytes = getIPv6ByName(ip, ipv4Mapped);
+        if (bytes == null) {
+            return null;
+        }
+        try {
+            return Inet6Address.getByAddress(null, bytes, -1);
+        } catch (UnknownHostException e) {
+            throw new RuntimeException(e); // Should never happen
+        }
+    }
+
+    /**
+     * Returns the byte array representation of a {@link CharSequence} IP address.
+     * <p>
+     * The {@code ipv4Mapped} parameter specifies how IPv4 addresses should be treated.
+     * "IPv4 mapped" format as
+     * defined in <a href="http://tools.ietf.org/html/rfc4291#section-2.5.5">rfc 4291 section 2</a> is supported.
+     * @param ip {@link CharSequence} IP address to be converted to a {@link Inet6Address}
+     * @param ipv4Mapped
+     * <ul>
+     * <li>{@code true} To allow IPv4 mapped inputs to be translated into {@link Inet6Address}</li>
+     * <li>{@code false} Consider IPv4 mapped addresses as invalid.</li>
+     * </ul>
+     * @return byte array representation of the {@code ip} or {@code null} if not a valid IP address.
+     */
+    private static byte[] getIPv6ByName(CharSequence ip, boolean ipv4Mapped) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getIPv6ByName" is 94 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)